File: 02_algorithm/programmers/Level3/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [False] * n
    for i in range(n):
        if not visited[i]:
            def DFS(num):
                for idx in range(n):
                    if not visited[idx] and idx != num:
                        if computers[num][idx]:
                            visited[idx] = True
                            DFS(idx)
            visited[i] = True
            DFS(i)
            answer += 1
    return answer

print(solution(3, [[1, 1, 0], [1, 1, 1], [0, 1, 1]]))
print(solution(3, [[1, 1, 0], [1, 1, 0], [0, 0, 1]])

File: 06_js/04_algorithm/programmers/Level3/네트워크.js
function solution(n, computers) {
  let answer = 0;
  let visited = Array(n).fill().map(_ => false);
  
  const DFS = num => {
    for (let j = 0; j < n; j++) {
      if (!visited[j] && j !== num && computers[num][j]) {
        visited[j] = true;
        DFS(j);
      }
    }
  };
  
  for (let i = 0; i < n; i++) {
    if (!visited[i]) {
      visited[i] = true;
      DFS(i);
      answer += 1;
    }
  }
  return answer;
}

console.log(solution(3, [[1, 1, 0], [1, 1, 0], [0, 0, 1]])); // 2
console.log(solution(3, [[1, 1, 0], [1, 1, 1], [0, 1, 1]])); // 1